dist/maven-antrun/build-parallel-worlds.xml

```diff
@@ -122,15 +122,34 @@
             <fileset dir="${project.build.directory}/parallel-world" includes="spark*/${shimServiceRsrc}"/>
         </concat>
 
+        <exec executable="${project.basedir}/scripts/binary-dedupe.sh"
```

Comment:
What's the rationale of executing binary-dedupe.sh first? Previous iteration of the PR was what we wanted IMO. Now we allow again the build to fail on binary compatibility check whereas it's not the root cause in https://github.com/NVIDIA/spark-rapids/pull/4265#issuecomment-986072583

Reply:
Updated